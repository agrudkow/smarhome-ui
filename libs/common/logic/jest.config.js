module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
