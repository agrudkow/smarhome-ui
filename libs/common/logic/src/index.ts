export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
