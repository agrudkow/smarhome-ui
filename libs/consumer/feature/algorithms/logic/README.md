# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
