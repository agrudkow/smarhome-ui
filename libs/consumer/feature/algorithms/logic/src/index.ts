export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
export * from './lib/use-algorithms-list';
export * from './lib/use-algorithm-detials';
export * from './lib/use-algorithm-rating';
export * from './lib/use-select-algorithm';
