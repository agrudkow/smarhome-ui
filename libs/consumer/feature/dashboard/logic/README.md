# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
