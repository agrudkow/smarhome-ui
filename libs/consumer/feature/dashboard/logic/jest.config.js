module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
