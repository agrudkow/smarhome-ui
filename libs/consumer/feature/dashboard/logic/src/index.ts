export * from './lib/use-billings';
