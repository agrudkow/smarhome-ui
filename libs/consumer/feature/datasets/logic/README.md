# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
