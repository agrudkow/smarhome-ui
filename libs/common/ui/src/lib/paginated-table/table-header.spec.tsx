import React from 'react';
import TableHeader, { Order, Cell } from './table-header';
import { customRender } from '@smarthome/common/logic';

const cells: Cell<'test'>[] = [
  {
    disablePadding: false,
    id: 'test',
    align: 'center',
    label: 'Test',
  },
  {
    disablePadding: false,
    id: 'test',
    align: 'center',
    label: 'Test',
  },
];

describe(' TableHeader', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <table>
        <TableHeader<'test'>
          order={Order.asc}
          orderBy="test"
          cells={cells}
          onRequestSort={jest.fn()}
        />
      </table>
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot when expanded', () => {
    const { baseElement } = customRender(
      <table>
        <TableHeader<'test'>
          order={Order.asc}
          orderBy="test"
          cells={cells}
          onRequestSort={jest.fn()}
        />
      </table>
    );
    expect(baseElement.firstChild).toMatchSnapshot();
  });
});
