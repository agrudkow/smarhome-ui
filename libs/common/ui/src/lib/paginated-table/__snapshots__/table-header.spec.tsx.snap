// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` TableHeader should match snapshot when expanded 1`] = `
<div>
  <table>
    <thead
      class="MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort="ascending"
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
          scope="col"
        >
          <b>
            Test
          </b>
        </th>
        <th
          aria-sort="ascending"
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
          scope="col"
        >
          <b>
            Test
          </b>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
